plugins/common/utils_test.go

```diff
@@ -153,3 +153,93 @@ func TestCreateHeaders(t *testing.T) {
 		})
 	}
 }
+
+func Test_getHostAndPortFromTargetURL(t *testing.T) {
+	type args struct {
+		url              string
+		defaultNamespace string
+	}
+	tests := []struct {
+		name     string
+		args     args
+		wantHost string
+		wantPort string
+	}{
+		{
+			name: "no port",
+			args: args{
+				url: "http://catalogue.sock-shop",
+			},
+			wantHost: "catalogue.sock-shop",
+			wantPort: "80",
+		},
+		{
+			name: "with port",
+			args: args{
+				url: "http://catalogue.sock-shop:8080",
+			},
+			wantHost: "catalogue.sock-shop",
+			wantPort: "8080",
+		},
+		{
+			name: "with port - no namespace - use default",
```

Comment:
need to add a case for external host `www.example.com` with `defaultNamespace` set